src/components/select-field/BaseSelectField.js

```diff
@@ -336,20 +337,21 @@ class BaseSelectField extends React.Component<Props, State> {
         }
     };
 
-    getItemFromFilteredOptions = (index: number) => {
+    getFilteredOptions = () => {
         const { options } = this.props;
         const { searchText } = this.state;
 
-        const filteredOptions = options.filter(option =>
-            searchText ? option.displayText.toLowerCase().includes(searchText.toLowerCase()) : true,
-        );
+        return options.filter(option => {
+            const isSubstring = option.displayText.toLowerCase().includes(searchText.toLowerCase());
+            const isClearOption = option.value === CLEAR;
```

Comment:
As requested by design, we should be filtering out the Clear Option as part of the options when we're performing searching.